dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -140,6 +140,8 @@
 		<!-- Verify DI trimmability at development-time, but turn the validation off for production/trimmed builds. -->
 		<VerifyDependencyInjectionOpenGenericServiceTrimmability Condition="'$(VerifyDependencyInjectionOpenGenericServiceTrimmability)' == '' And '$(_BundlerDebug)' != 'true'">false</VerifyDependencyInjectionOpenGenericServiceTrimmability>
 		<VerifyDependencyInjectionOpenGenericServiceTrimmability Condition="'$(VerifyDependencyInjectionOpenGenericServiceTrimmability)' == ''">true</VerifyDependencyInjectionOpenGenericServiceTrimmability>
+		<!-- This should be set by dotnet/sdk instead, once https://github.com/dotnet/sdk/issues/25392 gets resolved.  -->
+		<DynamicCodeSupport Condition="'$(DynamicCodeSupport)' == '' And '$(UseInterpreter)' != 'true' And ('$(_PlatformName)' == 'iOS' Or '$(_PlatformName)' == 'tvOS' Or '$(_PlatformName)' == 'watchOS')">false</DynamicCodeSupport>
```

Comment:
I believe we should disable DynamicCodeSupport when both of the following are true:

1. We're using the AOT compiler, or we're on a platform where it's not possible to publish without the AOT compiler.
2. We're not using the interpreter.

Also ideally the debug configuration would work the same way as the release configuration, so that customers aren't surprised when they go to publish their apps.

This means that for the first point:

* Include iOS and tvOS.
* Include Mac Catalyst when building for arm64 (x64 has the JIT). This is slightly problematic though, since by default we build for arm64 only for release builds (i.e. the debug build doesn't necessarily know that the release build uses a different runtime identifier). As such, I think it would make sense to handle Mac Catalyst as if arm64 is required and disable dynamic code support even when only building for x64. If someone wants to publish a Mac Catalyst app on x64 only, then can force DynamicCodeSupport=true themselves.

Note: we don't support watchOS in .NET.

So:

```suggestion
		<DynamicCodeSupport Condition="'$(DynamicCodeSupport)' == '' And '$(UseInterpreter)' != 'true' And ('$(_PlatformName)' == 'iOS' Or '$(_PlatformName)' == 'tvOS' Or '$(_PlatformName)' == 'MacCatalyst')">false</DynamicCodeSupport>
```